src/ioctls/vcpu.rs

```diff
@@ -233,6 +235,75 @@ impl VcpuFd {
         Ok(())
     }
 
+    /// Specifies set of signals that are blocked during execution of KVM_RUN.
+    /// Signals that are not blocked will cause KVM_RUN to return with -EINTR.
+    ///
+    /// # Arguments
+    ///
+    /// * `sigmask` - Signal mask. For details check the `kvm_signal_mask` structure in the
+    ///             [KVM API doc](https://www.kernel.org/doc/Documentation/virtual/kvm/api.txt).
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # extern crate kvm_ioctls;
+    /// # extern crate vmm_sys_util;
+    /// # extern crate libc;
+    /// # use kvm_ioctls::Kvm;
+    /// # use vmm_sys_util::signal;
+    /// # use libc::sigset_t;
+    /// # use libc::*;
+    /// let kvm = Kvm::new().unwrap();
+    /// let vm = kvm.create_vm().unwrap();
+    /// let vcpu = vm.create_vcpu(0).unwrap();
+    /// let sigset = signal::create_sigset(&[SIGINT, SIGKILL, SIGTERM]).unwrap();
+    /// vcpu.set_signal_mask(&sigset);
+    /// ```
+    ///
+    pub fn set_signal_mask(&self, sigset: &sigset_t) -> Result<()> {
+        // C struct of 'kvm_signal_mask' is in form of:
+        //   struct kvm_signal_mask {
+        //       __u32 len;
+        //       __u8  sigset[0];
+        //   };
+        // The payload array 'sigset' doesn't take any space in the struct.
+        // In Rust we construct a vector of 'kvm_signal_mask' with continuous space,
```

Comment:
This looks like it could be implemented with a [FAM struct](https://github.com/rust-vmm/vmm-sys-util/blob/master/src/fam.rs). Here's an example: https://github.com/rust-vmm/kvm-ioctls/blob/f372a9d046aa741860799e7003e65996d7734103/src/ioctls/system.rs#L235